테스트씬, 카메라 변경, 스택, 엔딩메니저

yeongco opened the pull request:
에셋들 적용 안되어있을겁니다
코드 위주로 한번 읽어보세요
interaction코드에도 stack_manager 연결해줘야합니다. 연결했을떄 가동하는건 테스트 완료했습니다

jaewonjung6446 committed:
인벤토리 저장 기능

1

jaewonjung6446 committed:
Merge branch 'Develope' into jaewon

jaewonjung6446 committed:
텍스트 출력 상호작용

jaewonjung6446 committed:
텍스트출력 버그수정

jaewonjung6446 committed:
조명 깜빡임, 기타 조명 베이크

jaewonjung6446 committed:
빛 테스트

jaewonjung6446 committed:
빛 조절

jaewonjung6446 committed:
1

jaewonjung6446 committed:
상호작용 리팩토링 및 시점 전환

대상과 상호작용 시 시점이 자동으로 전환되는 메소드 완료. 버그... 는 아직 못 발견

jaewonjung6446 committed:
상호작용 리팩토링 및 씬 전환 기능 추가

1.상호작용 리팩토링: 이전에 Update상에서 모두 처리하던 것을 조건을 만족시키면 코루틴을 호출하는 방식으로 바꿈. 실시간 자원은 조금 더 잡아먹겠지만, 유지보수가 쉬울 것 같아서 리팩토링.
2.씬 전환 기능 추가: 추가라고 할 것도 없긴 한데... 그냥 특정 오브젝트 ray로 인식하면 씬이 전환되는 기능임.
3.구현 예정: 씬 전환에서 부드럽게 lerp로 전환하는 기능(실제로는 그렇게 보이게만 하는 것이지만...)

jaewonjung6446 committed:
리팩토링

구와악

jaewonjung6446 committed:
버그 픽스, 새 버그 발견

입력 주기로 인한 텍스트 출력 코루틴 버그 해결
but, 코루틴이 실행되는 동안 다른 오브젝트가 인식되면 그 오브젝트도 같이 실행되는 버그 발견

jaewonjung6446 committed:
버그픽스 및 항우울제 기능 구현

구와악 이제 라디오 해야지

jaewonjung6446 committed:
.

.

jaewonjung6446 committed:
.

.

jaewonjung6446 committed:
튜토리얼 간단하게

jaewonjung6446 committed:
일시정지 메뉴

jaewonjung6446 committed:
Merge branch 'yph' into jaewon

jaewonjung6446 committed:
UI

jaewonjung6446 committed:
상호작용 public static화(공유 사용 목적)

jaewonjung6446 committed:
public 전환

jaewonjung6446 committed:
궤도 조준 기능 구현 시작

1.Tutorial스크립트에서 코루틴으로 bomb라는 씬 자체를 불러오고 바로 해당 씬의 오브젝트들을 바로 SetActive(false)하는 형태. 궤도 조준 오브젝트가 raycast_hit되면 bomb씬의 오브젝트를 Setactive(true)하는 방식으로 진행 할 예정
2.그냥 UI를 껐다가 키는 거랑 리소스 비교해서 효율성에 따라 구현 방식이 바뀔 수도 있을 것 같다... 가 아니라 낼 바꿀 예정. 지금 구현한 내용이 너무 리소스를 많이 먹고 효율적이지도 못한 것 같아서

yeongco committed:
테스트씬, 카메라 변경, 스택, 엔딩메니저

jaewonjung6446 reviewed (approved):
확인했습니다. 코루틴 오브젝트별로 분리시킬게요

jaewonjung6446 committed:
왠지 모르겠지만 오류해결

패키지 RE-IMPORT

jaewonjung6446 committed:
리팩토링 (기본 골자는 이게 진짜 마지막)

기본 상호작용 구성 구조
1.상호작용 메소드(InterAction)가 인터페이스에 있다.
2.상속 받은 클래스(오브젝트 이름과 이름이 같으면 편함)에서 인터페이스의 상호작용 메소드를 구현한다.(코루틴으로 바꿔도 됨)
3.InterAction_ctrl상에서 ray가 오브젝트랑 충돌하면 오브젝트랑 이름이 같은 클래스를 자동으로 불러서 그 안의 InterAction()함수역시 자동 호출한다.(위기능을 하는 메소드의 자세한건 InterAction클래스에 기재)

jaewonjung6446 committed:
주석처리

.

jaewonjung6446 committed:
양식에 맞춰서 재 작업 중(항우울제)

jaewonjung6446 committed:
사진, 항우울제 리팩토링 완료

jaewonjung6446 committed:
Merge branch 'TestBranch_jaewon' into Testbranch_Dev

jaewonjung6446 committed:
그림 출력 오류 수정

jaewonjung6446 committed:
Merge branch 'YG' into TestBranch_jaewon

jaewonjung6446 committed:
미니게임 제작 시작

UI패널 생성, ESC시 게임 취소 기능(일시 정지랑 안 겹치게) 이제 게임 만들면 됨

jaewonjung6446 committed:
폭격 카운트 다운

코루틴으로 카운트 다운 함수 구현, Start에서 이를 바로 실행, 설명 끝나면 코루틴 내에서 카운트 다운 텍스트 들어가게 할 예정

jaewonjung6446 committed:
.

.

yeongco committed:
stack manager 수정

남 우울, 약물
여 우울, 집착 생성
추후 해당 내용으로 엔딩메니저 적용 예정

yeongco committed:
엔딩씬, 스택, 게임엔딩 설정

현재 우울증약 먹으면 엔딩씬으로 넘어가게 테스터 역할 부여해놨습니다
스택메니저(스택관리)->엔딩메니저(스택을 통한 엔딩결정)->엔딩(결정된 엔딩 보여주기)
형식의 연관관계 가지고있습니다

yeongco committed:
Merge pull request #7 from jaewonjung6446/YG_TestBranch

Yg test branch

yeongco committed:
Update Yeonggyo_test.unity

yeongco committed:
Merge pull request #8 from jaewonjung6446/YG_TestBranch

Update Yeonggyo_test.unity
